src/MongoDB/ReadPreference.c

```diff
@@ -465,6 +562,19 @@ static PHP_METHOD(ReadPreference, serialize)
 		ADD_ASSOC_LONG_EX(&retval, "maxStalenessSeconds", maxStalenessSeconds);
 	}
 
+	if (!bson_empty0(hedge)) {
+		php_phongo_bson_state state;
+
+		PHONGO_BSON_INIT_DEBUG_STATE(state);
```

Comment:
Serialization would also benefit from using a document (even more so than `get_properties`) to guarantee type fidelity when round-tripping. I don't expect this is an issue for the current structure of the `hedge` document but it could become one down the line.